### PR TITLE
fix data race by replace clone

### LINKED ISSUE #6230: Data race in `TestPreparingProgress`
### Which jobs are failing
```
==================
WARNING: DATA RACE
Write at 0x00c00b837458 by goroutine 178389:
  github.com/tikv/pd/pkg/core/storelimit.(*limit).Reset()
      /home/runner/work/pd/pd/pkg/core/storelimit/store_limit.go:133 +0x3f5
  github.com/tikv/pd/pkg/core/storelimit.(*StoreRateLimit).Reset()
      /home/runner/work/pd/pd/pkg/core/storelimit/store_limit.go:107 +0x8a
  github.com/tikv/pd/pkg/schedule.(*OperatorController).getOrCreateStoreLimit()
      /home/runner/work/pd/pd/pkg/schedule/operator_controller.go:870 +0x26c
  github.com/tikv/pd/pkg/schedule.(*OperatorController).exceedStoreLimitLocked()
      /home/runner/work/pd/pd/pkg/schedule/operator_controller.go:847 +0x69b
  github.com/tikv/pd/pkg/schedule.(*OperatorController).PromoteWaitingOperator()
      /home/runner/work/pd/pd/pkg/schedule/operator_controller.go:355 +0x510
  github.com/tikv/pd/pkg/schedule.(*OperatorController).AddWaitingOperator()
      /home/runner/work/pd/pd/pkg/schedule/operator_controller.go:314 +0x10d1
  github.com/tikv/pd/server/cluster.(*coordinator).runScheduler()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:809 +0x56d
  github.com/tikv/pd/server/cluster.(*coordinator).addScheduler.func2()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:639 +0x47

Previous read at 0x00c00b837458 by goroutine 178493:
  github.com/tikv/pd/pkg/core/storelimit.(*limit).Available()
      /home/runner/work/pd/pd/pkg/core/storelimit/store_limit.go:139 +0x148
  github.com/tikv/pd/pkg/core/storelimit.(*StoreRateLimit).Available()
      /home/runner/work/pd/pd/pkg/core/storelimit/store_limit.go:90 +0x8a
  github.com/tikv/pd/pkg/core.(*StoreInfo).IsAvailable()
      /home/runner/work/pd/pd/pkg/core/store.go:143 +0x236
  github.com/tikv/pd/pkg/schedule/filter.(*StoreStateFilter).exceedRemoveLimit()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:420 +0xb6
  github.com/tikv/pd/pkg/schedule/filter.(*StoreStateFilter).exceedRemoveLimit-fm()
      <autogenerated>:1 +0x29
  github.com/tikv/pd/pkg/schedule/filter.(*StoreStateFilter).anyConditionMatch()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:514 +0xae6
  github.com/tikv/pd/pkg/schedule/filter.(*StoreStateFilter).Source()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:530 +0x199
  github.com/tikv/pd/pkg/schedule/filter.SelectSourceStores.func1.1()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:37 +0x10b
  github.com/tikv/pd/pkg/slice.AllOf[...].func1()
      /home/runner/work/pd/pd/pkg/slice/slice.go:35 +0x4b
  github.com/tikv/pd/pkg/slice.AnyOf[...]()
      /home/runner/work/pd/pd/pkg/slice/slice.go:20 +0xc1
  github.com/tikv/pd/pkg/slice.NoneOf[...]()
      /home/runner/work/pd/pd/pkg/slice/slice.go:29 +0xb8
  github.com/tikv/pd/pkg/slice.AllOf[...]()
      /home/runner/work/pd/pd/pkg/slice/slice.go:37 +0xf8
  github.com/tikv/pd/pkg/schedule/filter.SelectSourceStores.func1()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:36 +0x133
  github.com/tikv/pd/pkg/schedule/filter.filterStoresBy()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:122 +0x216
  github.com/tikv/pd/pkg/schedule/filter.SelectSourceStores()
      /home/runner/work/pd/pd/pkg/schedule/filter/filters.go:35 +0xb9
  github.com/tikv/pd/pkg/schedule/schedulers.(*balanceWitnessScheduler).Schedule()
      /home/runner/work/pd/pd/pkg/schedule/schedulers/balance_witness.go:241 +0xa89
  github.com/tikv/pd/server/cluster.(*scheduleController).Schedule()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:899 +0x31d
  github.com/tikv/pd/server/cluster.(*coordinator).runScheduler()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:808 +0x50a
  github.com/tikv/pd/server/cluster.(*coordinator).addScheduler.func2()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:639 +0x47

Goroutine [1783](https://github.com/tikv/pd/actions/runs/4530971783/jobs/7980551492#step:5:1784)89 (running) created at:
  github.com/tikv/pd/server/cluster.(*coordinator).addScheduler()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:639 +0x3b5
  github.com/tikv/pd/server/cluster.(*coordinator).run()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:403 +0x2e7d
  github.com/tikv/pd/server/cluster.(*coordinator).runUntilStop()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:306 +0x75
  github.com/tikv/pd/server/cluster.(*RaftCluster).runCoordinator()
      /home/runner/work/pd/pd/server/cluster/cluster.go:576 +0x11b
  github.com/tikv/pd/server/cluster.(*RaftCluster).Start.func2()
      /home/runner/work/pd/pd/server/cluster/cluster.go:295 +0x39

Goroutine [1784](https://github.com/tikv/pd/actions/runs/4530971783/jobs/7980551492#step:5:1785)93 (running) created at:
  github.com/tikv/pd/server/cluster.(*coordinator).addScheduler()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:639 +0x3b5
  github.com/tikv/pd/server/cluster.(*coordinator).run()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:403 +0x2e7d
  github.com/tikv/pd/server/cluster.(*coordinator).runUntilStop()
      /home/runner/work/pd/pd/server/cluster/coordinator.go:306 +0x75
  github.com/tikv/pd/server/cluster.(*RaftCluster).runCoordinator()
      /home/runner/work/pd/pd/server/cluster/cluster.go:576 +0x11b
  github.com/tikv/pd/server/cluster.(*RaftCluster).Start.func2()
      /home/runner/work/pd/pd/server/cluster/cluster.go:295 +0x39
==================
```
### CI link
https://github.com/tikv/pd/actions/runs/4530971783/jobs/7980551492

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #6230

### What is changed and how does it work?
using `ResetLimit` to set limit rate. The `ResetLimit ` will use lock

```commit-message
```

### Check List

Tests

- Unit test
```
func TestDataRace(t *testing.T) {
	store := NewStoreInfo(&metapb.Store{Id: uint64(1)}, ResetStoreLimit(storelimit.AddPeer, 10))
	wg := sync.WaitGroup{}
	wg.Add(1)
	go func() {
		store.GetStoreLimit().Reset(20, storelimit.AddPeer)
		wg.Done()
	}()
	store.IsAvailable(storelimit.AddPeer)
	wg.Wait()
}
```

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/operator_controller.go [hunk 1]
```diff
@@ -866,7 +866,8 @@ func (oc *OperatorController) getOrCreateStoreLimit(storeID uint64, limitType st
 		return nil
 	}
 
-	limit := s.GetStoreLimit()
-	limit.Reset(ratePerSec, limitType)
-	return limit
+	if limit := s.GetStoreLimit(); limit.Cap(limitType) != ratePerSec {
+		oc.cluster.GetBasicCluster().ResetStoreLimit(storeID, limitType, ratePerSec)
+	}
+	return s.GetStoreLimit()
```
Comment (rleungx) [inline pkg/schedule/operator_controller.go:872]: How about just changing back and not adding a new interface?

**Review by lhy1024 — COMMENTED** [review]
Do we need to add the test of pr description in the test file?

**Comment by bufferflies** [comment]
> Do we need to add the test of pr description in the test file?

I think none, it is caused by unlock.

File: pkg/schedule/operator_controller.go (hunk 1 above)
Comment (bufferflies) [reply pkg/schedule/operator_controller.go:872]: `ResetStoreLimit` will clone first. So if it doesn't compare the cap, it will waste `clone` once.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**